src/Day1.hs

```diff
@@ -130,16 +130,21 @@ updateGameOver input st = st
     isStart = lookupKey (keys input) Enter == Pressed || lookupKey (keys input) (Char ' ') == Pressed
 
 updatePlayState :: Input -> PlayState -> PlayState
-updatePlayState input ps = case S.member nextPos wPos of  
-  True  -> ps
-  False -> ps { player = p { playerPos = updatePlayerPos input (playerPos p) } }
+updatePlayState input pState
+  | S.member nextPos ws = pState
+  | isJust $ lookupMap currPos ds = case lookupMap currPos ds of
```

Comment:
* remove this conditition
* move the right hand side of the` lookupMap` to otherwise; because `Nothing -> pState` will never be reached. 
* Replace that line ^ with `pState { player = p { playerPos = updatePlayerPos input (playerPos p) } }`